lib/simplifyExpression/basicsSearch/basicRules.js

```diff
@@ -0,0 +1,224 @@
+const {build, query} = require('math-nodes');
+const {rewriteNode} = require('math-rules');
+
+const ChangeTypes = require('../../ChangeTypes');
+const checks = require('../../checks');
+const defineRuleString = require('../../util/defineRuleString');
+const Node = require('../../node');
+
+// e.g. -(-3) -> 3
+const NEGATION = defineRuleString('--#a', '#a');
```

Comment:
(we could also just collapse multiple of these steps in a row somewhere else)

alsooo I'd call this RESOLVE_DOUBLE_NEGATION because negation sounds like you're negating it